packages/kbn-es/src/install/source.js

```diff
@@ -0,0 +1,67 @@
+const childProcess = require('child_process');
+const path = require('path');
+const chalk = require('chalk');
+const simpleGit = require('simple-git');
+const { installArchive } = require('./archive');
+const { findMostRecentlyChanged, log } = require('../utils');
+const { GRADLE_BIN, ES_ARCHIVE_PATTERN, BASE_PATH } = require('../paths');
+
+/**
+ * Installs ES from source
+ *
+ * @param {Object} options
+ * @property {String} options.installPath
+ * @property {String} options.sourcePath
+ */
+exports.installSource = async function installSource({
+  sourcePath,
+  basePath = BASE_PATH,
+  installPath = path.resolve(basePath, 'source'),
+}) {
+  const branchName = await getBranchName(sourcePath);
+
+  log.info('source path: %s', chalk.bold(sourcePath));
+  log.info('install path: %s', chalk.bold(installPath));
+  log.info('on branch %s', chalk.bold(branchName));
+
+  const archive = await createSnapshot({ sourcePath });
+
+  await installArchive(archive, { basePath, installPath });
+
+  return { installPath };
+};
+
+/**
+ * Creates archive from source
+ *
+ * @param {Object} options
+ * @property {String} options.sourcePath
+ * @returns {Object} containing archive and optional plugins
+ */
+async function createSnapshot({ sourcePath }) {
+  const buildArgs = [':distribution:archives:tar:assemble'];
+
+  childProcess.execFileSync(GRADLE_BIN, buildArgs, {
```

Comment:
Mind using `await execa()` instead of execFileSync? It's just as easy, is async, and takes care to kill the child procs when the process exits.

Reply:
27557268a0e3f5f35f7f8373a50ca3e6cdbe867c